### PR TITLE
[feature/hidden-files] Add support for showing/hiding hidden items

### PR DESCRIPTION
**Pull request by felix-schwarz**
## Description
Adds support for showing/hiding hidden items (items starting with a .):
- added new ownCloudApp.framework to share code between app and file provider
	- DisplaySettings class keeps track of settings, change notifications and can update queries (acting as filter for them)
- added new "Display Settings" settings section with "Show hidden files and folders" option (using DisplaySettings)
- added support for DisplaySettings to File Provider
- changed ProgressSummarizer summaries from "Performing [remainCount] of [totalCount] .." to "Performing [remainCount] .." (forgotten in concurrency branch)
- cleaned up SettingsViewController
- removed GlobalSettingsViewController demo class
- switched to lastest ios-sdk and updated app code to use it (mostly .isRootPath)

## Related Issue
#390 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test plan:

https://github.com/owncloud/QA/blob/master/Mobile/iOS-app/Hidden%20files.md

### CODE REVIEW

File: ownCloudAppFrameworkTests/ownCloudAppTests.m [hunk 1]
```diff
@@ -0,0 +1,37 @@
+//
```
Comment (mneuwert) [inline ownCloudAppFrameworkTests/ownCloudAppTests.m]: @felix-schwarz why did you add this?

File: ownCloudAppFramework/DisplaySettings.m [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		return (@{
+			OCClassSettingsKeyDisplayShowHiddenFiles : @(NO)
+		});
+	}
+
+	return (nil);
+}
+
+#pragma mark - Singleton
++ (DisplaySettings *)sharedDisplaySettings
```
Comment (mneuwert) [inline ownCloudAppFramework/DisplaySettings.m]: @felix-schwarz wouldn't it be better to call it just shared, since `[DisplaySesttings sharedDisplaySettings]` is too explicit for my taste at least?

**Review by mneuwert — CHANGES_REQUESTED** [review]

File: ownCloudAppFrameworkTests/ownCloudAppTests.m (hunk 1 above)
Comment (felix-schwarz) [reply ownCloudAppFrameworkTests/ownCloudAppTests.m]: `ownCloudApp.framework` will likely accumulate more shared code between app and File Provider over time (or Obj-C helpers for the app), at which point unit tests may become relevant. I wasn't sure how easy it is to add unit tests for a target later, so I chose to add them right away.

File: ownCloudAppFramework/DisplaySettings.m (hunk 2 above)
Comment (felix-schwarz) [reply ownCloudAppFramework/DisplaySettings.m]: This is to match the naming conventions (i.e. `NSWorkspace.sharedWorkspace`, `NSAppleEventManager.sharedAppleEventManager`, etc.). It will be translated to `DisplaySettings.shared` in Swift automatically.

**Review by mneuwert — APPROVED** [review]
@felix-schwarz convinced:)

**Review by jesmrec — APPROVED** [review]
